flow/src/org/labkey/flow/controllers/editscript/UploadAnalysisForm.java

```diff
@@ -58,10 +60,26 @@ public void reset()
         }
     }
 
-    public void setWorkspaceObject(String object) throws Exception
+    // Stash the workspace in session and reference it via a token
+    public String getToken()
     {
-        _workspaceObject = (Workspace) PageFlowUtil.decodeObject(object);
+        if (_workspaceObject != null && token == null)
+        {
+            token = SessionHelper.stashAttribute(getRequest(), _workspaceObject, TimeUnit.MINUTES.toMillis(10));
+        }
+        return token;
+    }
+
+    // On form POST, get the workspace from session using the token
+    public void setToken(String token)
+    {
+        this.token = token;
+        if (token != null)
+        {
+            _workspaceObject = (Workspace)SessionHelper.getStashedAttribute(getRequest(), token);

```

Comment:
If token lookup fails the validate() method should generate a meaningful error message.